src/JustEat.StatsD/Buffered/StatsDUtf8Formatter.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace JustEat.StatsD.Buffered
+{
+    internal class StatsDUtf8Formatter
+    {
+        private readonly byte[] _utf8Prefix;
+
+        public StatsDUtf8Formatter(string prefix)
+        {
+            _utf8Prefix = string.IsNullOrWhiteSpace(prefix) ? new byte[0] : Encoding.UTF8.GetBytes(prefix + ".");
```

Comment:
What about:

```csharp
_utf8Prefix =
    string.IsNullOrWhiteSpace(prefix) ?
#if NET451
    new byte[0] :
#else
    Array.Empty<byte>() :
#endif
    Encoding.UTF8.GetBytes(prefix + ".");
```

Reply:
It is not a hot path anyway, created once per app lifetime

Reply:
OK, I just figured maybe it was worth doing if the PR is trying to reduce allocations.